src/app/recover-funds/[address]/RecoverFunds.tsx

```diff
@@ -0,0 +1,144 @@
+'use client';
+import { Address } from '@arbitrum/sdk';
+import { BigNumber, utils, constants } from 'ethers';
+import { useEffect, useState } from 'react';
+import { useNetwork } from 'wagmi';
+import { getProviderFromChainId, getTargetChainId } from '@/utils';
+import { supportedL1Networks } from '@/constants';
+import { RecoverFundsButton } from './RecoverFundsButton';
+
+interface OperationInfo {
+  balanceToRecover: BigNumber;
+  aliasedAddress: string;
+}
+
+const hasBalanceOverThreshold = (balanceToRecover: BigNumber) => {
+  // Aliased account will always have some leftover, we can't check for balance of 0, as it would always return 0
+  // We compare with 0.005 instead
+  return balanceToRecover.gte(BigNumber.from(5_000_000_000_000_000));
+};
+
+async function getData(
+  chainID: number,
+  address: string,
+): Promise<OperationInfo | null> {
+  // First, obtain the aliased address of the signer
+  const signerAddress = new Address(address);
```

Comment:
Something to keep in mind, this is actually not the `signerAddress`, but the address we've specified in the form. Maybe we should rename the variable to avoid future mistakes?

Reply:
Make sense, updated to `destinationAddress`